keps/sig-scheduling/5710-workload-aware-preemption/README.md

```diff
@@ -0,0 +1,1064 @@
+   - all running workloads with (preemption) priority lower than the new workload W
+   - all individual pods (not being part of workloads) with priority lower than the new workload W
+
+1. If removing all the potential victims would not make the new workload W schedulable,
+   the workload is unschedulable even with preemption.
+
+```
+<<[UNRESOLVED PodDisruptionBudget violations]>>
+How critical is reprieving workloads and pods violating PodDisruptionBudgets? We no longer can
+afford full workload scheduling trying to reprieve every individual pod and workload.
+
+We could consider finding the first one that can't be reprieved using binary search, but if we
+can't reprieve any of those, learning about that would require O(N) full workload schedulings
+with N being number of workload/pods violating PDB.
+<<[/UNRESOLVED]>>
+```
+
+1. For remaining potential victims, using binary search across priorities find the minimal priority P
+   for which scheduling the new workload W doesn't require preempting any workloads and/or pods with
+   priority higher than P. This allows to reduce the potential cascading preemptions later.
+
+```
+<<[UNRESOLVED minimizing preemptions]>>
+The following algorithm is by far no optimal, but is simple to reason about and I would suggest it as
+a starting point:
+- assume that all potential victims on the list are removed and schedule the new workload W
+- go over the remaining potential victims starting from the highest priority and check if these can
+  be placed in the place they are currently running; if so remove from the potential victims
+
+As a bonus we may consider few potential placements of the new workload W here and choose the one that
+somehow optimizes the number of victims. But that will become more critical once we get to
+Topology-Aware-Scheduling and I would leave that optimization until then.
+<<[/UNRESOLVED]>>
+```
+
+```
+<<[UNRESOLVED sharing algorithms]>>
+The remaining question is to what extent we want to unify the preemption mechanism across
+pod-triggerred (existing algorithm) and workload-triggerred preemption.
+
+It might be tempting to start with a dedicated new implementation to reduce the risk. But the above
+proposal was structured such way to facilitate sharing:
+- once the new workload W is placed, going over the remaining potential victims and trying to
+  place them where they are currently running, is pretty much exactly what the current algorithm is
+  doing
+- considering "few potential placements" in the pod-triggerred case can be used as "try every node"
+  so effectively it's also the existing algorithm (just viewed from a slightly different angle
+
+So I would actually argue to we should refactor the existing preemption code and use that in both
+cases.
+<<[/UNRESOLVED]
+```
+
+### Delayed preemption
+
+```
+<<[UNRESOLVED delayed preemption]>>
+Should we leave it as part of this KEP or should this be moved to the Gang-Scheduling one?
```

Comment:
I believe that it should be moved to another KEP, I feel that it is completely independent of workload aware preemption and can work just with the current preemption + gang scheduling.